src/modules/adaptors/Frax/DebtFTokenAdaptorV2.sol

```diff
@@ -0,0 +1,300 @@
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure virtual override returns (bytes32) {
+        return keccak256(abi.encode("FraxLend debtToken Adaptor V 1.0"));
+    }
+
+    //============================================ Implement Base Functions ===========================================
+
+    /**
+     * @notice User deposits are NOT allowed into this position.
+     */
+    function deposit(uint256, bytes memory, bytes memory) public pure override {
+        revert BaseAdaptor__UserDepositsNotAllowed();
+    }
+
+    /**
+     * @notice User withdraws are NOT allowed from this position.
+     */
+    function withdraw(uint256, address, bytes memory, bytes memory) public pure override {
+        revert BaseAdaptor__UserWithdrawsNotAllowed();
+    }
+
+    /**
+     * @notice This position is a debt position, and user withdraws are not allowed so
+     *         this position must return 0 for withdrawableFrom.
+     */
+    function withdrawableFrom(bytes memory, bytes memory) public pure override returns (uint256) {
+        return 0;
+    }
+
+    /**
+     * @notice Returns the cellar's balance of the respective Fraxlend debtToken calculated from cellar borrow shares
+     * @param adaptorData encoded fraxlendPair (fToken) for this position
+     */
+    function balanceOf(bytes memory adaptorData) public view override returns (uint256) {
+        IFToken fraxlendPair = abi.decode(adaptorData, (IFToken));
+        return _toBorrowAmount(fraxlendPair, fraxlendPair.userBorrowShares(msg.sender), false, ACCOUNT_FOR_INTEREST);
+    }
+
+    /**
+     * @notice Returns `assetContract` from respective fraxlend pair, but this is most likely going to be FRAX.
+     */
+    function assetOf(bytes memory adaptorData) public view override returns (ERC20) {
+        IFToken fraxlendPair = abi.decode(adaptorData, (IFToken));
+        return ERC20(fraxlendPair.asset());
+    }
+
+    /**
+     * @notice This adaptor reports values in terms of debt.
+     */
+    function isDebt() public pure override returns (bool) {
+        return true;
+    }
+
+    //============================================ Strategist Functions ===========================================
+
+    // `borrowAsset`
+    /**
+     * @notice Allows strategists to borrow assets from Fraxlend.
+     * @notice `debtTokenToBorrow` must be the debtToken, NOT the underlying ERC20.
+     * @param fraxlendPair the Fraxlend Pair to borrow from.
+     * @param amountToBorrow the amount of `debtTokenToBorrow` to borrow on Fraxlend.
+     * NOTE: `borrowAsset` is the same btw v1 and v2 FraxlendPairs
+     */
+    function borrowFromFraxlend(IFToken fraxlendPair, uint256 amountToBorrow) public {
+        bytes32 positionHash = keccak256(abi.encode(identifier(), true, abi.encode(address(fraxlendPair))));
```

Comment:
Use `_validateFToken` here instead of this block of code

Reply:
Done